pkg/hubble/metrics/drop/handler_test.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package drop
+
+import (
+	"context"
+	"testing"
+
+	pb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/hubble/metrics/api"
+	monitorAPI "github.com/cilium/cilium/pkg/monitor/api"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDropHandler(t *testing.T) {
+	registry := prometheus.NewRegistry()
+	opts := api.Options{"sourceContext": "namespace", "destinationContext": "namespace"}
+
+	dropHandler := &dropHandler{}
+
+	t.Run("Init", func(t *testing.T) {
+		require.NoError(t, dropHandler.Init(registry, opts))
+	})
+
+	t.Run("Status", func(t *testing.T) {
+		require.Equal(t, "destination=namespace,source=namespace", dropHandler.Status())
+	})
+
+	t.Run("ProcessFlow_ShouldReportNothingForForwardedFlow", func(t *testing.T) {
+		flow := &pb.Flow{
+			EventType: &pb.CiliumEventType{Type: monitorAPI.MessageTypePolicyVerdict},
+			L4: &pb.Layer4{
+				Protocol: &pb.Layer4_TCP{
+					TCP: &pb.TCP{},
+				},
+			},
+			Source:      &pb.Endpoint{Namespace: "foo"},
+			Destination: &pb.Endpoint{Namespace: "bar"},
+			Verdict:     pb.Verdict_FORWARDED,
+		}
+		dropHandler.ProcessFlow(context.TODO(), flow)
+
+		metricFamilies, err := registry.Gather()
+		require.NoError(t, err)
+		require.Empty(t, metricFamilies)
+
+	})
+
+	t.Run("ProcessFlow_ShouldReportDroppedFlow", func(t *testing.T) {
+		flow := &pb.Flow{
+			EventType: &pb.CiliumEventType{Type: monitorAPI.MessageTypePolicyVerdict},
+			L4: &pb.Layer4{
+				Protocol: &pb.Layer4_TCP{
+					TCP: &pb.TCP{},
+				},
+			},
+			Source:         &pb.Endpoint{Namespace: "foo"},
+			Destination:    &pb.Endpoint{Namespace: "bar"},
+			Verdict:        pb.Verdict_DROPPED,
+			DropReason:     uint32(pb.DropReason_POLICY_DENIED),
+			DropReasonDesc: pb.DropReason_POLICY_DENIED,
+		}
+		dropHandler.ProcessFlow(context.TODO(), flow)
+
+		metricFamilies, err := registry.Gather()
+		require.NoError(t, err)
+
+		assert.Equal(t, "hubble_drop_total", *metricFamilies[0].Name)
+		require.Len(t, metricFamilies[0].Metric, 1)
+		metric := metricFamilies[0].Metric[0]
```

Comment:
You never inspect the value of the metric itself, only the labels. I think you can assert against `metric.Counter.Value`